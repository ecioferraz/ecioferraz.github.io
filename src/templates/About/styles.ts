import styled, { css } from 'styled-components';

export const Container = styled.main`
  ${({ theme }) => css`
    @media ${theme.media.width.gteMedium} or (orientation: landscape) {
      flex-direction: row;
      justify-content: space-evenly;

      > * {
        margin: 0 ${theme.spacings.large};
      }
    }


    img {
      border-radius: ${theme.spacings.large};
      box-sizing: border-box;
      max-width: 20rem;

      @media ${theme.media.height.lteXSmall} and (orientation: landscape) {
        width: 12rem;
      }
    }
  `}
`;

export const AboutContainer = styled.section`
  ${({ theme }) => css`
    max-width: 50rem;

    h1 {
      margin-bottom: ${theme.spacings.xsmall};
    }

    p {
      text-align: justify;
    }

    @media ${theme.media.height.lteXSmall} and (orientation: landscape) {
      h1 {
        font-size: ${theme.font.sizes.medium};
        margin: 0;
      }

      p {
        font-size: ${theme.font.sizes.xsmall};
      }
    }
  `}
`;
