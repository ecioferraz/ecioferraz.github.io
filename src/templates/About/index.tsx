import { HandPeace } from '@styled-icons/fa-solid';
import ImageCard from '../../components/ImageCard';
import TextCard from '../../components/TextCard';
import { theme } from '../../styles/theme';
import * as Styled from './styles';
import hero from '../../../public/assets/images/hero/hero.jpg';

export default function About() {
  return (
    <Styled.Container id="about">
      <ImageCard alt="Picture of Écio" className="slideInDown" src={hero} />
      <Styled.AboutContainer>
        <TextCard className="slideInRight">
          Olá! <HandPeace size={theme.font.sizes.xlarge} />
        </TextCard>
        <TextCard as="p" className="slideInRight" size="small">
          Me chamo Écio, sou de Recife, Pernambuco. Sempre fui curioso e gostei
          de procurar soluções para os problemas que me apareciam com
          tecnologia, então decidi me tornar desenvolvedor web através da Trybe.
          Esse é meu portfólio, e aqui você pode conferir os projetos e
          habilidades que tenho desenvolvido.
        </TextCard>
      </Styled.AboutContainer>
    </Styled.Container>
  );
}
