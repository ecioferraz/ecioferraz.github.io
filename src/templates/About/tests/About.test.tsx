import renderTheme from '../../../styles/renderTheme';
import About from '..';

describe('<About />', () => {
  it('should render <About />', () => {
    const { container } = renderTheme(<About />);

    expect(container).toMatchSnapshot();
  });
});
