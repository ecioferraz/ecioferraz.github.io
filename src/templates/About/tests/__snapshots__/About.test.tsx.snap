// Vitest Snapshot v1

exports[`<About /> > should render <About /> 1`] = `
.c4 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c1 {
  width: 100%;
}

.c3 {
  font-size: 5.6rem;
}

.c5 {
  font-size: 1.6rem;
}

.c0 img {
  border-radius: 3.2rem;
  box-sizing: border-box;
  max-width: 20rem;
}

.c2 {
  max-width: 50rem;
}

.c2 h1 {
  margin-bottom: 0.8rem;
}

.c2 p {
  text-align: justify;
}

@media (max-width:768px) {
  .c3 {
    font-size: 4.8rem;
  }
}

@media (min-width:768px) or (orientation:landscape) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-box-pack: space-evenly;
    -webkit-justify-content: space-evenly;
    -ms-flex-pack: space-evenly;
    justify-content: space-evenly;
  }

  .c0 > * {
    margin: 0 3.2rem;
  }
}

@media (max-height:375px) and (orientation:landscape) {
  .c0 img {
    width: 12rem;
  }
}

@media (max-height:375px) and (orientation:landscape) {
  .c2 h1 {
    font-size: 2.4rem;
    margin: 0;
  }

  .c2 p {
    font-size: 0.9rem;
  }
}

<div>
  <main
    class="c0"
    id="about"
  >
    <img
      alt="Picture of Écio"
      class="c1"
      src="../assets/images/hero/hero.jpg"
    />
    <section
      class="c2"
    >
      <h1
        class="c3"
      >
        Olá! 
        <svg
          aria-hidden="true"
          class="c4"
          fill="currentColor"
          focusable="false"
          height="4.0rem"
          viewBox="0 0 512 512"
          width="4.0rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224 0c17.7 0 32 14.3 32 32v208h-64V32c0-17.7 14.3-32 32-32zm96 160c17.7 0 32 14.3 32 32v64c0 17.7-14.3 32-32 32s-32-14.3-32-32v-64c0-17.7 14.3-32 32-32zm64 64c0-17.7 14.3-32 32-32s32 14.3 32 32v64c0 17.7-14.3 32-32 32s-32-14.3-32-32v-64zM93.3 51.2 175.9 240h-69.8L34.7 76.8c-7.1-16.2.3-35 16.5-42.1s35.1.3 42.1 16.5zm27 221.3-.2-.5H216c22.1 0 40 17.9 40 40s-17.9 40-40 40h-56c-8.8 0-16 7.2-16 16s7.2 16 16 16h56c39.8 0 72-32.2 72-72v-.6c9.4 5.4 20.3 8.6 32 8.6 13.2 0 25.4-4 35.6-10.8 8.7 24.9 32.5 42.8 60.4 42.8 11.7 0 22.6-3.1 32-8.6v8.6c0 88.4-71.6 160-160 160h-61.7c-42.4 0-83.1-16.9-113.1-46.9l-11.6-11.6C77.5 429.5 64 396.9 64 363v-27c0-32.7 24.6-59.7 56.3-63.5z"
            fill="currentColor"
          />
        </svg>
      </h1>
      <p
        class="c5"
      >
        Me chamo Écio, sou de Recife, Pernambuco. Sempre fui curioso e gostei de procurar soluções para os problemas que me apareciam com tecnologia, então decidi me tornar desenvolvedor web através da Trybe. Esse é meu portfólio, e aqui você pode conferir os projetos e habilidades que tenho desenvolvido.
      </p>
    </section>
  </main>
</div>
`;
